tests/shared/src/test/scala/cats/tests/AlgebraInvariantSuite.scala

```diff
@@ -179,11 +179,27 @@ class AlgebraInvariantSuite extends CatsSuite with ScalaVersionSpecificAlgebraIn
   implicit private val arbCommutativeGroupInt: Arbitrary[CommutativeGroup[Int]] =
     Arbitrary(genCommutativeGroupInt)
 
+  protected val integralForMiniInt: Numeric[MiniInt] with Integral[MiniInt] = new MiniIntNumeric
+    with Integral[MiniInt] {
+    def quot(x: MiniInt, y: MiniInt): MiniInt = MiniInt.unsafeFromInt(x.toInt / y.toInt)
+    def rem(x: MiniInt, y: MiniInt): MiniInt = MiniInt.unsafeFromInt(x.toInt % y.toInt)
+  }
+
+  // This is a spurious instance since MiniInt is not a Fractional data type. But we use it since we don't have a
+  // Fractional type for which ExhaustiveCheck is also implemented. See https://github.com/typelevel/cats/pull/4033
+  protected val fractionalForMiniInt: Numeric[MiniInt] with Fractional[MiniInt] = new MiniIntNumeric
+    with Fractional[MiniInt] {
+    def div(x: MiniInt, y: MiniInt): MiniInt =
+      if (y == MiniInt.zero) {
+        MiniInt.maxValue
+      } else {
+        x / y
+      }
+  }
+
   implicit private val arbNumericMiniInt: Arbitrary[Numeric[MiniInt]] = Arbitrary(Gen.const(integralForMiniInt))
   implicit private val arbIntegralMiniInt: Arbitrary[Integral[MiniInt]] = Arbitrary(Gen.const(integralForMiniInt))
-  implicit private val arbFractionalFloat: Arbitrary[Fractional[Float]] = Arbitrary(
-    Gen.const(implicitly[Fractional[Float]])
-  )
+  implicit private val arbFractionalFloat: Arbitrary[Fractional[MiniInt]] = Arbitrary(Gen.const(fractionalForMiniInt))
```

Comment:
Since this is a spurious instance, probably better to pass it explicitly than have it floating around in implicit scope. Also would be good to add a comment to that effect.
```suggestion
  private val arbFractionalMiniInt: Arbitrary[Fractional[MiniInt]] = Arbitrary(Gen.const(fractionalForMiniInt))
```

Reply:
Alternatively, maybe define it only at it's use site? e.g.
```scala
{
  implicit private val arbFractionalMiniInt = ???
  checkAll("Invariant[Fractional]", InvariantTests[Fractional].invariant[MiniInt, Boolean, Boolean])
}